shm/inc/map.hpp

```diff
@@ -7,9 +7,11 @@
 
 class Map {
 public:
+    Map();
```

Comment:
What does it do?

Reply:
It's a constructor. You have its definition in cpp file